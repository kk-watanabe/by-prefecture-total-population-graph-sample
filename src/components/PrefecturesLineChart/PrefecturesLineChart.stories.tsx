import { PrefecturesLineChart } from "./PrefecturesLineChart";

import type { ComponentStory, ComponentMeta } from "@storybook/react";

export default {
  title: "PrefecturesLineChart",
  component: PrefecturesLineChart,
} as ComponentMeta<typeof PrefecturesLineChart>;

const Template: ComponentStory<typeof PrefecturesLineChart> = (args) => <PrefecturesLineChart {...args} />;

const data = [
  {
    year: "1960年",
    1: 4000,
    47: 2400,
  },
  {
    year: "1970年",
    1: 3000,
    47: 1398,
  },
  {
    year: "1980年",
    1: 2000,
    47: 9800,
  },
  {
    year: "1990年",
    1: 2780,
    47: 3908,
  },
  {
    year: "2000年",
    1: 1890,
    47: 4800,
  },
  {
    year: "2010年",
    1: 2390,
    47: 3800,
  },
  {
    year: "2020年",
    1: 3490,
    47: 4300,
  },
];

const selectedPrefectures = [
  {
    prefCode: 1,
    prefName: "北海道",
  },
  {
    prefCode: 47,
    prefName: "沖縄県",
  },
];

export const Default = Template.bind({});
Default.args = {
  height: 600,
  data,
  selectedPrefectures,
};
