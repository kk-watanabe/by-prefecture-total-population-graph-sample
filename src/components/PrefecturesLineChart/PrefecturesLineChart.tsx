import { ResponsiveContainer, LineChart, CartesianGrid, Line, Label, XAxis, YAxis, Tooltip, Legend } from "recharts";

import { Pref } from "@/api/prefectures/type";
import getPrefColor from "@/utils/prefectures-util";

export type PrefecturesLineChartData = {
  year: string;
  [key: number]: number;
};

export interface PrefecturesLineChartProps {
  height: number;
  data: PrefecturesLineChartData[];
  selectedPrefectures: Pref[];
}

export const PrefecturesLineChart = (props: PrefecturesLineChartProps) => (
  <ResponsiveContainer width="100%" height={props.height}>
    <LineChart data={props.data} margin={{ top: 80, right: 80, left: 10, bottom: 20 }}>
      <CartesianGrid />
      <XAxis dataKey="year" tickSize={15}>
        <Label value="年度" position="right" offset={40} />
      </XAxis>
      <YAxis tickSize={15}>
        <Label value="人口数" position="top" offset={40} />
      </YAxis>
      <Tooltip />
      <Legend wrapperStyle={{ bottom: 0 }} />
      {props.selectedPrefectures.map((p) => (
        <Line
          key={p.prefCode}
          type="monotone"
          dataKey={p.prefCode}
          name={p.prefName}
          stroke={getPrefColor(p.prefCode)}
        />
      ))}
    </LineChart>
  </ResponsiveContainer>
);
