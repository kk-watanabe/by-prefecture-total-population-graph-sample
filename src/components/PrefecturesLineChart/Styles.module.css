.customT-tooltip {
  padding: 16px;
  border: 1px solid var(--color-gray300);
  border-radius: 8px;
  background-color: var(--color-white);
  box-shadow: 2px 2px 4px rgba(0, 0, 0, 0.1);
}

.customT-tooltip__label {
  color: var(--color-gray700);
  font-weight: bold;
  font-size: var(--fs-sm);
}

.customT-tooltip__values {
  display: flex;
  flex-wrap: wrap;
  margin-top: 8px;
  max-width: 280px;
  font-size: var(--fs-xs);

  gap: 8px;
}
