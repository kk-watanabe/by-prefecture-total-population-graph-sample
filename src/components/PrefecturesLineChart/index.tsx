/* istanbul ignore next */
export * from "./PrefecturesLineChart";
