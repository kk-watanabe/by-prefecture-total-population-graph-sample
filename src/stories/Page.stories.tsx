import React from "react";

import { Page } from "./Page";

import type { ComponentStory, ComponentMeta } from "@storybook/react";

export default {
  title: "Example/Page",
  component: Page,
  parameters: {
    // More on Story layout: https://storybook.js.org/docs/react/configure/story-layout
    layout: "fullscreen",
  },
} as ComponentMeta<typeof Page>;

const Template: ComponentStory<typeof Page> = (args) => <Page {...args} />;

export const LoggedOut = Template.bind({});

export const LoggedIn = Template.bind({});
