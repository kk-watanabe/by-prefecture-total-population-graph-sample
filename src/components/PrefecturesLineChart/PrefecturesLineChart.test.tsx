import { composeStories } from "@storybook/testing-react";
import { render, waitFor } from "@testing-library/react";

import * as stories from "./PrefecturesLineChart.stories";

const { Default } = composeStories(stories);

test("Readner Default components", async () => {
  const { container } = render(<Default />);

  await waitFor(
    () => {
      expect(container.firstChild).toMatchSnapshot();
    },
    {
      interval: 10000,
    }
  );
});
