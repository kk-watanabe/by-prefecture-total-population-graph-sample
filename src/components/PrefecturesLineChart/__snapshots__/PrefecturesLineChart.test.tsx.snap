// Vitest Snapshot v1

exports[`Readner Default components 1`] = `
<div
  class="recharts-responsive-container"
  style="width: 100%; height: 600px;"
/>
`;
